app/src/main/java/com/infomaniak/mail/utils/SharedUtils.kt

```diff
@@ -104,6 +108,14 @@ class SharedUtils @Inject constructor(
         }
     }
 
+    fun updateAiFeatureFlag() {
+        with(ApiRepository.checkAiFeatureFlag()) {
+            if (isSuccess()) {
+                featureFlagController.upsertFeatureFlag(FeatureFlag(FeatureFlagType.AI, data == true))
+            }
+        }
+    }
```

Comment:
```suggestion
    fun updateAiFeatureFlag() = with(ApiRepository.checkAiFeatureFlag()) {
            if (isSuccess())   featureFlagController.upsertFeatureFlag(FeatureFlag(FeatureFlagType.AI, isEnabled = data == true))
    }
```

Reply:
Apparently we don't want to do this in this specific case?

Reply:
Why ? There's no difference when we use `with(binding)` or `with(navargs)` 🤔 
At least could we oneline this if

Reply:
Didn't notice the if inlining, I did it